packages/nuxt/src/core/plugins/layer-aliasing.ts

```diff
@@ -0,0 +1,69 @@
+import { createUnplugin } from 'unplugin'
+import type { NuxtConfigLayer } from 'nuxt/schema'
+import { resolveAlias } from '@nuxt/kit'
+import { normalize } from 'pathe'
+import MagicString from 'magic-string'
+
+interface LayerAliasingOptions {
+  sourcemap?: boolean
+  transform?: boolean
+  layers: NuxtConfigLayer[]
+}
+
+const ALIAS_RE = /(?<=['"])[~@]{1,2}(?=\/)/g
+
+export const LayerAliasingPlugin = createUnplugin((options: LayerAliasingOptions) => {
+  const aliases = Object.fromEntries(options.layers.map(l => [l.config.srcDir || l.cwd, {
+    '~': l.config.srcDir || l.cwd,
+    '@': l.config.srcDir || l.cwd,
+    '~~': l.config.rootDir || l.cwd,
+    '@@': l.config.rootDir || l.cwd,
+    ...l.config?.alias || {}
+  }]))
```

Comment:
Passing aliases to `resolveAlias`, removes all user (top level) aliases. We might merge it with user aliases on `nuxt.options.alias` (maybe lower priority but also for me makes sense that it has higher priority than layer aliases to give users an escape hatch)

Reply:
Then there is no way to for example force resolve a library to specific version. But lets try and iterate.

Reply:
it would still work. the layer probably is not providing a resolution for that library, right?

Reply:
Layer might also use aliases for resolution.

Reply:
I think we can actually just list it inline to allow layer to configure it, and that way we avoid the issue.